core/src/avm2/object/xml_object.rs

```diff
@@ -251,12 +255,49 @@ impl<'gc> TObject<'gc> for XmlObject<'gc> {
                     return Err(format!("Cannot set attribute {:?} without a local name", name).into());
                 };
                 let value = value.coerce_to_string(activation)?;
-                let new_attr = E4XNode::attribute(activation.context.gc_context, local_name, value);
+                let new_attr = E4XNode::attribute(mc, local_name, value);
                 attributes.push(new_attr);
                 return Ok(());
             }
+
+            return Err(format!(
+                "Adding a new attribute to XML object is not yet implemented: {name:?} = {value:?}"
+            )
+            .into());
+        }
+
+        if value.as_object().map_or(false, |obj| {
+            obj.as_xml_object().is_some() || obj.as_xml_list_object().is_some()
+        }) {
+            return Err(
+                format!("Cannot set an XML/XMLList object {value:?} as an element yet").into(),
+            );
+        }
+
+        if name.is_any_name() {
+            return Err("Any name (*) not yet implemented for set".into());
         }
 
-        Err(format!("Modifying an XML object is not yet implemented: {name:?} = {value:?}").into())
+        let text = value.coerce_to_string(activation)?;
+
+        let matches: Vec<_> = children
```

Comment:
I realize this is temp first-pass code, but would be good to avoid allocating here and use the iterator directly when we implement replacing multiple elems.